connect/runtime/src/main/java/org/apache/kafka/connect/runtime/SourceTaskOffsetCommitter.java

```diff
@@ -67,70 +69,53 @@ public void close(long timeoutMs) {
     public void schedule(final ConnectorTaskId id, final WorkerSourceTask workerTask) {
         synchronized (committers) {
             long commitIntervalMs = config.getLong(WorkerConfig.OFFSET_COMMIT_INTERVAL_MS_CONFIG);
-            ScheduledFuture<?> commitFuture = commitExecutorService.schedule(new Runnable() {
+            ScheduledFuture<?> commitFuture = commitExecutorService.scheduleWithFixedDelay(new Runnable() {
                 @Override
                 public void run() {
                     commit(id, workerTask);
                 }
-            }, commitIntervalMs, TimeUnit.MILLISECONDS);
-            committers.put(id, new ScheduledCommitTask(commitFuture));
+            }, 0, commitIntervalMs, TimeUnit.MILLISECONDS);
+            committers.put(id, commitFuture);
```

Comment:
I think I missed this in the last round, but if we're using `synchronized (committers)` here anyway, what's the value in using the `ConcurrentHashMap`? In fact, if we're relying on this synchronization to protect against concurrent modification in the `remove` and `commit` cases (the latter being the one I am concerned with since you could schedule the commit and have the commit task execute before it is added to `committers`), then I don't think we're properly protected atm since the lock used by `ConcurrentHashMap` is not the same as the normal monitor for the object.